### PR TITLE
First cut at OpenIdConnectConfiguration and SignedHttpRequest

### PR DESCRIPTION
**Pull request by brentschmaltz**
These changes remove the use of Newtonsoft from OpenIdConnectConfiguration and OpenIdConnectMessage completely.
SignedHttpRequestHandler still has some dependency on usage in JsonWebTokenHandler. That will be next.

The serialization of OpenIdConnectConfiguration follows the pattern of JsonWebKey and JsonWebKeySet.
SignedHttpRequestHandler changes were mostly related to using Utf8JsonReader/Writers all the through the stack rather than creating temporary strings that need to be transformed back utf8bytes.
This led to a number of changes in the SignedHttpRequestHandler tests.

The tests that checked for a null payload were no longer needed as the temporary dictionary is not used anymore.

### CODE REVIEW

File: src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectParameterNames.cs [hunk 1]
```diff
@@ -51,6 +53,56 @@ public static class OpenIdConnectParameterNames
...
         public const string VersionTelemetry = "x-client-ver";
-#pragma warning restore 1591
     }
+
+    /// <summary>
+    /// Parameter names for OpenIdConnect UTF8 bytes.
+    /// </summary>
+    public static class OpenIdConnectParameterUtf8Bytes
+    {
+        public static readonly byte[] AccessToken = Encoding.UTF8.GetBytes("access_token");
```
Comment (keegan-caruso) [inline src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectParameterNames.cs]: see this comment: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/2208#discussion_r1288896116

Can we use ReadOnlySpan<byte> and u8 strings?

File: src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectParameterNames.cs [hunk 2]
```diff
@@ -51,6 +53,56 @@ public static class OpenIdConnectParameterNames
         public const string UserId = "user_id";
         public const string Username = "username";
         public const string VersionTelemetry = "x-client-ver";
-#pragma warning restore 1591
     }
+
+    /// <summary>
+    /// Parameter names for OpenIdConnect UTF8 bytes.
+    /// </summary>
+    public static class OpenIdConnectParameterUtf8Bytes
```
Comment (keegan-caruso) [inline src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectParameterNames.cs]: does this need to be public or can it be internal?

File: src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdProviderMetadataNames.cs [hunk 3]
```diff
@@ -56,6 +58,60 @@ public static class OpenIdProviderMetadataNames
...
+    }
 #pragma warning restore 1591
+
+    /// <summary>
+    /// OpenIdProviderConfiguration MetadataName - UTF8Bytes
+    /// http://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata
+    /// </summary>
+    internal static class OpenIdProviderMetadataUtf8Bytes
+    {
+        public static readonly byte[] AcrValuesSupported = Encoding.UTF8.GetBytes("acr_values_supported");
```
Comment (keegan-caruso) [inline src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdProviderMetadataNames.cs]: same question, can we use ReadOnlySpan and u8 strings?

File: src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestHandler.cs [hunk 4]
```diff
@@ -79,30 +82,78 @@ public string CreateSignedHttpRequest(SignedHttpRequestDescriptor signedHttpRequ
...
                 throw LogHelper.LogArgumentNullException(nameof(signedHttpRequestDescriptor));
 
             if (signedHttpRequestDescriptor.SigningCredentials == null)
                 throw LogHelper.LogArgumentNullException(nameof(signedHttpRequestDescriptor.SigningCredentials));
 
-            var payload = CreateHttpRequestPayload(signedHttpRequestDescriptor, callContext);
-            var header = new JObject();
-            if (signedHttpRequestDescriptor.AdditionalHeaderClaims != null && signedHttpRequestDescriptor.AdditionalHeaderClaims.Count != 0)
+            string encodedPayload;
+            using (MemoryStream payloadUtf8Bytes = new MemoryStream())
```
Comment (keegan-caruso) [inline src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestHandler.cs]: see: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/2208#discussion_r1288891375

Can we at a minimum use the memory streams buffer?

File: src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestHandler.cs [hunk 5]
```diff
@@ -79,30 +82,78 @@ public string CreateSignedHttpRequest(SignedHttpRequestDescriptor signedHttpRequ
...
+                try
+                {
+                    payloadWriter = new Utf8JsonWriter(payloadUtf8Bytes, new JsonWriterOptions { Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping });
+                    payloadWriter.WriteStartObject();
+
+                    CreateHttpRequestPayload(ref payloadWriter, signedHttpRequestDescriptor);
+
+                    payloadWriter.WriteEndObject();
+                    payloadWriter.Flush();
+                    encodedPayload = Base64UrlEncoder.Encode(payloadUtf8Bytes.ToArray());
```
Comment (keegan-caruso) [inline src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestHandler.cs]: same comment as above can we use the streams buffer?

File: src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestHandler.cs [hunk 6]
```diff
@@ -114,201 +165,195 @@ public string CreateSignedHttpRequest(SignedHttpRequestDescriptor signedHttpRequ
...
+                try
+                {
+                    writer = new Utf8JsonWriter(memoryStream, new JsonWriterOptions { Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping });
+                    writer.WriteStartObject();
+
+                    CreateHttpRequestPayload(ref writer, signedHttpRequestDescriptor);
+
+                    writer.WriteEndObject();
+                    writer.Flush();
+                    return Encoding.UTF8.GetString(memoryStream.ToArray());
```
Comment (keegan-caruso) [inline src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestHandler.cs]: same comment on buffer

File: src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestHandler.cs [hunk 7]
```diff
@@ -1307,7 +1349,7 @@ private Uri EnsureAbsoluteUri(Uri uri)
         /// https://datatracker.ietf.org/doc/html/draft-ietf-oauth-signed-http-request-03#section-4.1
         /// https://datatracker.ietf.org/doc/html/draft-ietf-oauth-signed-http-request-03#section-7.5
         /// </remarks>
-        private static Dictionary<string, string> SanitizeHeaders(IDictionary<string, IEnumerable<string>> headers)
+        private static IDictionary<string, string> SanitizeHeaders(IDictionary<string, IEnumerable<string>> headers)
```
Comment (keegan-caruso) [inline src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestHandler.cs:1329]: it is a private method, why return an interface?

File: src/Microsoft.IdentityModel.Protocols/AuthenticationProtocolMessage.cs [hunk 8]
```diff
@@ -20,7 +20,7 @@ public abstract class AuthenticationProtocolMessage
         private string _scriptButtonText = "Submit";
         private string _scriptDisabledText = "Script is disabled. Click Submit to continue.";
 
-        private Dictionary<string, string> _parameters = new Dictionary<string, string>();
+        private IDictionary<string, string> _parameters = new Dictionary<string, string>();
```
Comment (keegan-caruso) [inline src/Microsoft.IdentityModel.Protocols/AuthenticationProtocolMessage.cs:23]: why change to an interface here?

File: src/Microsoft.IdentityModel.JsonWebTokens/JwtTokenUtilities.cs [hunk 9]
```diff
@@ -85,6 +85,7 @@ public static string CreateEncodedSignature(string input, SigningCredentials sig
         /// <exception cref="ArgumentNullException"><paramref name="input"/> or <paramref name="signingCredentials"/> is null.</exception>
         public static string CreateEncodedSignature(string input, SigningCredentials signingCredentials, bool cacheProvider)
         {
+            // TODO create overload that takes a Span<byte> for the input
```
Comment (jennyf19) [inline src/Microsoft.IdentityModel.JsonWebTokens/JwtTokenUtilities.cs:88]: >TODO create overload that takes a Span<byte> for the input [](http://example.com/codeflow?start=15&length=58)

do we need a github issue to track this?

File: src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectParameterNames.cs (hunk 1 above)
Comment (jennyf19) [reply src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectParameterNames.cs]: nice catch Keegan.

File: src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Configuration/OpenIdConnectConfiguration.cs [hunk 10]
```diff
@@ -67,242 +96,350 @@ public OpenIdConnectConfiguration()
...
+        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+#endif
         public string EndSessionEndpoint { get; set; }
 
         /// <summary>
         /// Gets or sets the 'frontchannel_logout_session_supported'.
         /// </summary>
-        [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore, NullValueHandling = NullValueHandling.Ignore, PropertyName = OpenIdProviderMetadataNames.FrontchannelLogoutSessionSupported, Required = Required.Default)]
+        /// <remarks>Would be breaking to change, in 6x it was string, spec says bool.
+        /// TODO - add another property, obsolete and drop in 8x?
```
Comment (jennyf19) [inline src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Configuration/OpenIdConnectConfiguration.cs:209]: > TODO - add another property, obsolete and drop in 8x? [](http://example.com/codeflow?start=11&length=54)

do we have a GitHub issue tracking this?

File: src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Json/OpenIdConnectConfigurationSerializer.cs [hunk 11]
```diff
@@ -0,0 +1,615 @@
...
+            "ID_TOKEN_ENCRYPTION_ALG_VALUES_SUPPORTED",
+            "ID_TOKEN_ENCRYPTION_ENC_VALUES_SUPPORTED",
+            "ID_TOKEN_SIGNING_ALG_VALUES_SUPPORTED",
+            "INTROSPECTION_ENDPOINT",
+            "INTROSPECTION_ENDPOINT_AUTH_METHODS_SUPPORTED",
+            "INTROSPECTION_ENDPOINT_AUTH_SIGNING_ALG_VALUES_SUPPORTED",
+            "JWKS_URI",
+            "ISSUER",
+            "LOGOUT_SESSION_SUPPORTED",
+            // TODO - should this be a first class property?
```
Comment (jennyf19) [inline src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Json/OpenIdConnectConfigurationSerializer.cs]: >should this be a first class property [](http://example.com/codeflow?start=22&length=37)

what do you mean?

File: src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Json/OpenIdConnectConfigurationSerializer.cs [hunk 12]
```diff
@@ -0,0 +1,615 @@
...
+                        LogHelper.MarkAsNonPII(reader.CurrentDepth),
+                        LogHelper.MarkAsNonPII(reader.BytesConsumed))));
+
+            while(JsonPrimitives.ReaderRead(ref reader))
+            {
+                #region Check property name using ValueTextEquals
+                // the config spec, https://datatracker.ietf.org/doc/html/rfc7517#section-4, does not require that we reject JSON with
+                // duplicate member names, in strict mode, we could add logic to try a property once and throw if a duplicate shows up.
+                // 6x uses the last value.
+                // TODO - With collections, make sure two properties are not additive
```
Comment (jennyf19) [inline src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Json/OpenIdConnectConfigurationSerializer.cs:116]: >With collections, make sure two properties are not additive [](http://example.com/codeflow?start=26&length=59)

GH issue?

File: src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Json/OpenIdConnectConfigurationSerializer.cs [hunk 13]
```diff
@@ -0,0 +1,615 @@
...
+                        JsonPrimitives.ReadStrings(ref reader, config.ScopesSupported, MetadataName.ScopesSupported, ClassName, true);
+
+                    else if (reader.ValueTextEquals(Utf8Bytes.ServiceDocumentation))
+                        config.ServiceDocumentation = JsonPrimitives.ReadString(ref reader, MetadataName.ScopesSupported, ClassName, true);
+
+                    else if (reader.ValueTextEquals(Utf8Bytes.SubjectTypesSupported))
+                        JsonPrimitives.ReadStrings(ref reader, config.SubjectTypesSupported, MetadataName.SubjectTypesSupported, ClassName, true);
+
+                    else if (reader.ValueTextEquals(Utf8Bytes.SubjectTypesSupported))
+                        JsonPrimitives.ReadStrings(ref reader, config.SubjectTypesSupported, MetadataName.SubjectTypesSupported, ClassName, true);
```
Comment (jennyf19) [inline src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Json/OpenIdConnectConfigurationSerializer.cs:245]: are these two duplicates?

File: src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Json/OpenIdConnectConfigurationSerializer.cs [hunk 14]
```diff
@@ -0,0 +1,615 @@
...
+                                JsonPrimitives.ReadStrings(ref reader, config.ScopesSupported, MetadataName.ScopesSupported, ClassName, true);
+
+                            else if (propertyName.Equals(MetadataName. ServiceDocumentation, StringComparison.OrdinalIgnoreCase))
+                                config.ServiceDocumentation = JsonPrimitives.ReadString(ref reader, MetadataName.ScopesSupported, ClassName, true);
+
+                            else if (propertyName.Equals(MetadataName. SubjectTypesSupported, StringComparison.OrdinalIgnoreCase))
+                                JsonPrimitives.ReadStrings(ref reader, config.SubjectTypesSupported, MetadataName.SubjectTypesSupported, ClassName, true);
+
+                            else if (propertyName.Equals(MetadataName. SubjectTypesSupported, StringComparison.OrdinalIgnoreCase))
+                                JsonPrimitives.ReadStrings(ref reader, config.SubjectTypesSupported, MetadataName.SubjectTypesSupported, ClassName, true);
```
Comment (jennyf19) [inline src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Json/OpenIdConnectConfigurationSerializer.cs]: duplicate?

File: src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Json/OpenIdConnectConfigurationSerializer.cs [hunk 15]
```diff
@@ -0,0 +1,615 @@
...
+        public static string Write(OpenIdConnectConfiguration OpenIdConnectConfiguration)
+        {
+            using (MemoryStream memoryStream = new MemoryStream())
+            {
+                Utf8JsonWriter writer = null;
+                try
+                {
+                    writer = new Utf8JsonWriter(memoryStream, new JsonWriterOptions { Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping });
+                    Write(ref writer, OpenIdConnectConfiguration);
+                    writer.Flush();
```
Comment (jennyf19) [inline src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Json/OpenIdConnectConfigurationSerializer.cs:461]: >writer [](http://example.com/codeflow?start=20&length=6)

`writer?.Flush()` ?

File: src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestHandler.cs [hunk 16]
```diff
@@ -114,201 +165,195 @@ public string CreateSignedHttpRequest(SignedHttpRequestDescriptor signedHttpRequ
...
+        /// Adds the 'm' claim to the <paramref name="writer"/>.
         /// </summary>
-        /// <param name="payload">HttpRequest payload represented as a <see cref="Dictionary{TKey, TValue}"/>.</param>
+        /// <param name="writer"><see cref="Utf8JsonWriter"/></param>
         /// <param name="signedHttpRequestDescriptor">A structure that wraps parameters needed for SignedHttpRequest creation.</param>
         /// <remarks>
         /// This method will be executed only if <see cref="SignedHttpRequestCreationParameters.CreateP"/> is set to <c>true</c>.
         /// </remarks>  
-        internal virtual void AddPClaim(Dictionary<string, object> payload, SignedHttpRequestDescriptor signedHttpRequestDescriptor)
+        internal virtual void AddPClaim(ref Utf8JsonWriter writer, SignedHttpRequestDescriptor signedHttpRequestDescriptor)
```
Comment (jennyf19) [inline src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestHandler.cs:287]: >AddPClaim( [](http://example.com/codeflow?start=30&length=10)

comment says to add 'm' claim but it's called `AddPClaim`?

File: test/Microsoft.IdentityModel.Protocols.OpenIdConnect.Tests/OpenIdConnectMessageTests.cs [hunk 17]
```diff
@@ -25,15 +26,17 @@ public void Constructors(OpenIdConnectMessageTheoryData theoryData)
             TestUtilities.WriteHeader($"{this}.Constructors", theoryData);
             var context = new CompareContext($"{this}.ReadMetadata, {theoryData.TestId}");
             OpenIdConnectMessage messageFromJson;
-            OpenIdConnectMessage messageFromJsonObj;
+            // TODO - these local variable has been left commented out as
```
Comment (jennyf19) [inline test/Microsoft.IdentityModel.Protocols.OpenIdConnect.Tests/OpenIdConnectMessageTests.cs:29]: >these local variable has been left commente [](http://example.com/codeflow?start=22&length=43)

do we have a GH issue tracking the work?

File: test/Microsoft.IdentityModel.Protocols.OpenIdConnect.Tests/Json/OpenIdConnectConfiguration6x.cs [hunk 18]
```diff
@@ -0,0 +1,671 @@
...
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using Microsoft.IdentityModel.Abstractions;
+using Microsoft.IdentityModel.Logging;
+using Microsoft.IdentityModel.Tokens;
+using Newtonsoft.Json;
+
+namespace Microsoft.IdentityModel.Protocols.OpenIdConnect.Json.Tests
+{
+    internal static class LogMessages
```
Comment (jennyf19) [inline test/Microsoft.IdentityModel.Protocols.OpenIdConnect.Tests/Json/OpenIdConnectConfiguration6x.cs:15]: is there a reason these are not in the product code and just in the tests?

File: src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Json/OpenIdConnectConfigurationSerializer.cs [hunk 19]
```diff
@@ -0,0 +1,615 @@
...
+                    return Encoding.UTF8.GetString(memoryStream.ToArray());
```
Comment (keegan-caruso) [inline src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Json/OpenIdConnectConfigurationSerializer.cs]: can we use the memory streams buffer here instead as well?

File: src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestHandler.cs [hunk 20]
```diff
@@ -79,30 +82,78 @@ public string CreateSignedHttpRequest(SignedHttpRequestDescriptor signedHttpRequ
...
+                    headerWriter.WriteString(JwtHeaderParameterNames.Alg, signedHttpRequestDescriptor.SigningCredentials.Algorithm);
+                    headerWriter.WriteString(JwtHeaderParameterNames.Typ, SignedHttpRequestConstants.TokenType);
+                    if (signedHttpRequestDescriptor.AdditionalHeaderClaims != null)
+                        foreach (string key in signedHttpRequestDescriptor.AdditionalHeaderClaims.Keys)
+                            headerWriter.WriteString(key, signedHttpRequestDescriptor.AdditionalHeaderClaims[key].ToString());
 
-            return message + "." + JwtTokenUtilities.CreateEncodedSignature(message, signedHttpRequestDescriptor.SigningCredentials, false);
+                    headerWriter.WriteEndObject();
+                    headerWriter.Flush();
+                    encodedHeader = Base64UrlEncoder.Encode(headerUtf8Bytes.ToArray());
```
Comment (keegan-caruso) [inline src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestHandler.cs]: one more spot it looks like we can use the streams buffer

File: src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectParameterNames.cs (hunk 1 above)
Comment (brentschmaltz) [reply src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectParameterNames.cs]: @keegan-caruso yes, that is what Stephen wants us to do

File: src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectParameterNames.cs (hunk 2 above)
Comment (brentschmaltz) [reply src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectParameterNames.cs]: It's a tossup, we make the parameters public and some users like that.

For now let's make it internal.

File: src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdProviderMetadataNames.cs (hunk 3 above)
Comment (brentschmaltz) [reply src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdProviderMetadataNames.cs]: Yep, that is what Stephen wants us to do.

File: src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestHandler.cs (hunk 4 above)
Comment (brentschmaltz) [reply src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestHandler.cs]: Yes, will make that change.

File: src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestHandler.cs (hunk 5 above)
Comment (brentschmaltz) [reply src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestHandler.cs]: Yep.

File: src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestHandler.cs (hunk 6 above)
Comment (brentschmaltz) [reply src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestHandler.cs]: Yep

File: src/Microsoft.IdentityModel.JsonWebTokens/JwtTokenUtilities.cs (hunk 9 above)
Comment (brentschmaltz) [reply src/Microsoft.IdentityModel.JsonWebTokens/JwtTokenUtilities.cs:88]: I have made the change in brentsch/JsonWebTokenHandler, which builds upon this.
I will remove the TODO.

File: test/Microsoft.IdentityModel.Protocols.OpenIdConnect.Tests/OpenIdConnectMessageTests.cs (hunk 17 above)
Comment (brentschmaltz) [reply test/Microsoft.IdentityModel.Protocols.OpenIdConnect.Tests/OpenIdConnectMessageTests.cs:29]: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/2224

File: src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestHandler.cs (hunk 4 above)
Comment (brentschmaltz) [reply src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestHandler.cs]: @keegan-caruso addressed in d789adbb30b8093a0eff70ee545983195935366f

File: src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestHandler.cs (hunk 6 above)
Comment (brentschmaltz) [reply src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestHandler.cs]: @keegan-caruso addressed in d789adbb30b8093a0eff70ee545983195935366f

File: src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestHandler.cs (hunk 5 above)
Comment (brentschmaltz) [reply src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestHandler.cs]: @keegan-caruso addressed in d789adbb30b8093a0eff70ee545983195935366f

File: src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Json/OpenIdConnectConfigurationSerializer.cs (hunk 13 above)
Comment (brentschmaltz) [reply src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Json/OpenIdConnectConfigurationSerializer.cs:245]: @jennyf19 addressed in d789adbb30b8093a0eff70ee545983195935366f

File: src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestHandler.cs (hunk 16 above)
Comment (brentschmaltz) [reply src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestHandler.cs:287]: @jennyf19 addressed in d789adbb30b8093a0eff70ee545983195935366f

File: src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Json/OpenIdConnectConfigurationSerializer.cs (hunk 19 above)
Comment (brentschmaltz) [reply src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Json/OpenIdConnectConfigurationSerializer.cs]: @keegan-caruso addressed in d789adbb30b8093a0eff70ee545983195935366f

File: src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestHandler.cs (hunk 20 above)
Comment (brentschmaltz) [reply src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestHandler.cs]: @keegan-caruso addressed in d789adbb30b8093a0eff70ee545983195935366f

File: test/Microsoft.IdentityModel.Protocols.OpenIdConnect.Tests/Json/OpenIdConnectConfiguration6x.cs (hunk 18 above)
Comment (brentschmaltz) [reply test/Microsoft.IdentityModel.Protocols.OpenIdConnect.Tests/Json/OpenIdConnectConfiguration6x.cs:15]: @jennyf19 It is a reminder that this is related to the Newtonsoft -> System.Text.Json conversion.
For the JsonWebTokenHandler, i realized we didn't need to change the name by adding 6x, we can just use FQN's

**Pull request merged**